### PR TITLE
修復「半形／全形」切換掣

### PR DESCRIPTION
**Pull request by skylee03**
而家「半形／全形字母及符號」及「半形／全形標點符號」切換掣只可以撳一次。應該係 `useLocalStorage` 嘅問題：
- streamich/react-use#2273
- streamich/react-use#2512

換成 `use-local-storage-state` （或者 `ahooks`）入邊嘅 `useLocalStorageState` 就解決到呢個問題。

### CODE REVIEW

**Comment by skylee03** [comment]
另外建議本倉庫啓用 issues 功能。

**Comment by graphemecluster** [comment]
唔該晒，不如順便將所有 `useLocalStorage` 嘅 occurrences 換晒做 `useLocalStorageState`？

> 另外建議本倉庫啓用 issues 功能。

我哋為咗方便集中管理，諗住叫想開 issues 嘅人喺 https://github.com/TypeDuck-HK/tracker 度開，呢個做法會唔會唔係太好？
又或者可以整個 CI 自動 transfer issues 嘅，你點睇？

**Comment by skylee03** [comment]
> 唔該晒，不如順便將所有 `useLocalStorage` 嘅 occurrences 換晒做 `useLocalStorageState`？

已經全部換晒。
 
> 我哋為咗方便集中管理，諗住叫想開 issues 嘅人喺 https://github.com/TypeDuck-HK/tracker 度開，呢個做法會唔會唔係太好？ 又或者可以整個 CI 自動 transfer issues 嘅，你點睇？

集中管理 issues 其實都幾好，但係而家網站入邊好似冇明確指引，好難揾到。😵‍💫

File: src/hooks.ts [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
 import type { PreferencesWithSetter } from "./types";
-import type { Dispatch, DispatchWithoutAction, SetStateAction } from "react";
+import type { DispatchWithoutAction } from "react";
 
-type UseLocalStorageOptions<T> = { raw: true } | { raw: false; serializer: (value: T) => string; deserializer: (value: string) => T } | undefined;
-export const useLocalStorage: {
-	<T>(key: string, initialValue: T, options?: UseLocalStorageOptions<T>): [T, Dispatch<SetStateAction<T>>, () => void];
-	<T>(key: string, initialValue?: T | undefined, options?: UseLocalStorageOptions<T>): [T | undefined, Dispatch<SetStateAction<T | undefined>>, () => void];
-} = _useLocalStorage;
+export const useLocalStorageState = _useLocalStorageState;
```
Comment (graphemecluster) [inline src/hooks.ts]: 不如 remove 咗呢行，唔好 re-export？

File: src/hooks.ts [hunk 2]
```diff
@@ -74,24 +69,14 @@ export function useRimeOption(option: string, defaultValue: boolean, deployStatu
...
-						deserializer(values) {
-							return new Set(values.split(",").map(value => value.trim() as Language));
-						},
-					}
-					: typeof value === "string"
-					? { raw: true }
-					: undefined,
+				{
+					defaultValue,
+					serializer: key === "displayLanguages" ? superjson : JSON,
```
Comment (graphemecluster) [inline src/hooks.ts]: 呢度係咪唔使 deserializer 嘅？
另外最後問下，如果要同現有嘅 storage backwards compatible，係咪要改做 `typeof value === "string" ? identity : JSON`？
或者另一個 approach 係我哋加個叫 `version` 嘅 key，如果冇就自動 migrate 去新格式？

Comment (skylee03) [reply src/hooks.ts]: > 呢度係咪唔使 deserializer 嘅？

`serializer` type 係 `{ stringify, parse }`，相當於包含晒 `useLocalStorage` 嘅 `serializer` 同 `deserializer`。

>  另外最後問下，如果要同現有嘅 storage backwards compatible，係咪要改做 `typeof value === "string" ? identity : JSON`？ 或者另一個 approach 係我哋加個叫 `version` 嘅 key，如果冇就自動 migrate 去新格式？

之前未考慮到 backwards compatible，我即刻 fix 吓。😅

**Comment by graphemecluster** [comment]
最後麻煩幫我 `dprint fmt` 一下（我知我應該加個 lint script 先啱，我之後會加）

**Comment by skylee03** [comment]
```
panic(main thread): index out of bounds
oh no: Bun has crashed. This indicates a bug in Bun, not your code.
```
- oven-sh/bun#10565

**Comment by skylee03** [comment]
> ```
> panic(main thread): index out of bounds
> oh no: Bun has crashed. This indicates a bug in Bun, not your code.
> ```
> 
> * [Bun crashed with index out of bounds oven-sh/bun#10565](https://github.com/oven-sh/bun/issues/10565)

This was fixed in oven-sh/bun#11763 and will be included in v1.1.13. We can try the fix in canary (`bun upgrade --canary`), or we can just wait for the release of the next version.

**Comment by graphemecluster** [comment]
果然用其他 package manager 都係有啲風險……
幫我 update 下個 `bun.lockb` 試下？

**Review by graphemecluster — APPROVED** [review]
要麻煩你做埋晒啲唔關事嘅嘢添，唔該晒！

**Pull request merged**